src/mapogcsld.cpp

```diff
@@ -4963,19 +4970,23 @@ static char *msSLDGetAttributeNameOrValue(const char *pszExpression,
     pszAttributeName = (char *)malloc(sizeof(char) * (nLength + 1));
     iValue = 0;
     for (i = 0; i < nLength - 2; i++) {
-      if ((pszExpression[i] != szCompare[0] ||
+      if ((pszExpression[i] != szCompare[0] &&
            pszExpression[i] != toupper(szCompare[0])) &&
-          (pszExpression[i] != szCompare2[0] ||
-           pszExpression[i] != toupper(szCompare2[0])))
+          (pszExpression[i] != szCompare2[0] &&
+           pszExpression[i] != toupper(szCompare2[0])) &&
+          (pszExpression[i] != szCompare3[0] &&
+           pszExpression[i] != toupper(szCompare3[0])))
 
       {
         pszAttributeName[iValue++] = pszExpression[i];
       } else {
         if (((pszExpression[i + 1] == szCompare[1] ||
               pszExpression[i + 1] == toupper(szCompare[1])) ||
              (pszExpression[i + 1] == szCompare2[1] ||
-              pszExpression[i + 1] == toupper(szCompare2[1]))) &&
-            (pszExpression[i + 2] == ' ')) {
+              pszExpression[i + 1] == toupper(szCompare2[1])) ||
```

Comment:
This (existing) logic seems flowed to me. It would possibly match a string that would be szCompare[0], szCompare2[1], szCompare3[2] (or similar permutations). So for example it could match "!>" (a mix of "!=" and "<>") Once we have selected which [0] character of szCompare/szCompare2/szCompare3, we should keep it for following comparisons.